src/CarbonAware.WebApi/test/unitTests/CarbonAwareControllerTests.cs

```diff
@@ -131,6 +133,56 @@ public async Task GetAverageCarbonIntensity_SuccessfulCallReturnsOk(string locat
         Assert.AreEqual(expectedContent, actualContent);
     }
 
+    /// <summary>
+    /// Tests that successfull call the average carbon intensity for a batch of requests on valid input
+    /// </summary>
+    [Test]
+    public async Task CalculateAverageCarbonIntensityBatch_ValidInput()
+    {
+        // Arrange
+        string location = "Sydney";
+        var start1 = new DateTimeOffset(2022, 3, 1, 0, 0, 0, TimeSpan.Zero);
+        var end1 = new DateTimeOffset(2022, 3, 1, 1, 0, 0, TimeSpan.Zero);
+
+        var start2 = new DateTimeOffset(2022, 3, 2, 0, 0, 0, TimeSpan.Zero);
+        var end2 = new DateTimeOffset(2022, 3, 2, 1, 0, 0, TimeSpan.Zero);
+        double data = 0.7;
+        var controller = new CarbonAwareController(this.MockCarbonAwareLogger.Object, CreateCarbonAwareAggregatorWithAverageCI(data).Object);
+
+        var request1 = new CarbonIntensityBatchDTO { Location = location, StartTime = start1, EndTime = end1 };
+        var request2 = new CarbonIntensityBatchDTO { Location = location, StartTime = start2, EndTime = end2 };
+        var requestList = new List<CarbonIntensityBatchDTO> { request1, request2 };
+        // Act
+        var actualContent = new List<CarbonIntensityDTO> { };
+        await foreach (var item in controller.GetAverageCarbonIntensityBatch(requestList))
+        {
+            actualContent.Add(item);
+        }
+
+        // Assert
+        var expectedContent1 = new CarbonIntensityDTO { Location = location, StartTime = start1, EndTime = end1, CarbonIntensity = data };
```

Comment:
well, the "pattern" in the code file is that all 'expect' vars are prepared in the last section, not in the //arrange, since they are not about setting up mocks (is my guess)